src/app/dashboard/dashboard/dashboard.component.ts

```diff
@@ -41,24 +43,81 @@ export class DashboardComponent implements OnInit, OnDestroy {
   dragulaSubscription = new Subscription();
   constructor(
     public teamService: TeamService,
-    private dragulaService: DragulaService,
+    private dragularService: DragulaService,
+    private allocationsService: AllocationsService,
+    private projectsService: ProjectsService,
+    private skillsService: SkillsService,
     private overlayService: OverlayService,
-    private studentsService: StudentsService
-  ) {
-    /* save model when modified by drag & drop operation */
-    this.dragulaSubscription.add(
-      dragulaService.dropModel('persons').subscribe(({ target, item }) => {
-        const newTeam = teamService.getTeamByName(target.id);
-        item.teamName = target.id;
-        // if person belongs to no Team (Person Pool) newTeam is undefined
-        newTeam?.add(item);
-        teamService.saveToLocalBrowserStorage();
-      })
+    private studentsService: StudentsService,
+    private studentPersonTransformerService: StudentToPersonService,
+    private projectTeamTransformerService: ProjectToTeamService
+  ) {}
+
+  private _students: Student[];
+  private _projects: Project[];
+  private _skills: Skill[];
+  private _allocations: Allocation[];
+
+  studentsWithoutTeam: Student[];
+  dataLoaded = false;
+
+  ngOnInit(): void {
+    this.studentsService.students$.subscribe(students => {
+      this._students = students;
+      this.updateAllocationsData();
+    });
+    this.allocationsService.allocations$.subscribe(allocations => {
+      this._allocations = allocations;
+      this.updateAllocationsData();
+    });
+    this.projectsService.projects$.subscribe(projects => {
+      this._projects = projects;
+      this.updateAllocationsData();
+    });
+    this.skillsService.skills$.subscribe(skills => {
+      this._skills = skills;
+      this.updateAllocationsData();
+    });
+    this.dragularService.drop('STUDENTS').subscribe(({ el, target, sibling }) => {
+      this.handleStudentDrop(el, target, sibling);
+    });
+  }
+
+  private handleStudentDrop(el: Element, target: Element, sibling: Element): void {
+    if (!el || !target) return;
+    const studentId = el.children[0].id;
+    const projectId = target.id;
+    const siblingId = sibling?.children[0].id;
+
+    if (!studentId) return;
+
+    if (!projectId) {
+      this.allocationsService.removeStudentFromProjects(studentId);
+      return;
+    }
+
+    this.allocationsService.moveStudentToProjectAtInset(studentId, projectId, siblingId);
+  }
+
+  private isDataLoaded(): boolean {
+    this.dataLoaded = !(
+      !this._students ||
```

Comment:
You don't need to check for the Array and it's length. It is sufficient if you only do that: 
```
this.dataLoaded = !(
      !this._students?.length ||
      !this._projects?.length ||
      !this._skills?.length ||
      !this._allocations?.length
    );
```